src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -84,31 +84,42 @@ internal partial class CodeFixService : ForegroundThreadAffinitizedObject, ICode
             using (var diagnostics = SharedPools.Default<List<DiagnosticData>>().GetPooledObject())
             {
                 var fullResult = await _diagnosticService.TryAppendDiagnosticsForSpanAsync(document, range, diagnostics.Object, cancellationToken: cancellationToken).ConfigureAwait(false);
-                foreach (var diagnostic in diagnostics.Object)
-                {
-                    cancellationToken.ThrowIfCancellationRequested();
+                var groupTasks = diagnostics.Object.GroupBy(d => d.Severity)
+                                        .Select(severityGroup =>
+                                            Task.Factory.SafeStartNewFromAsync(
+                                                () => GetFirstDiagnosticWithFix(document, severityGroup, range, cancellationToken), cancellationToken, TaskScheduler.Default));
 
-                    if (!range.IntersectsWith(diagnostic.TextSpan))
-                    {
-                        continue;
-                    }
+                var result = await Task.WhenAll(groupTasks).ConfigureAwait(false);
```

Comment:
we don't need to WhenAll here.  We can stop the moment we get the most severe severity back.